internal/products/api_base.go

```diff
@@ -0,0 +1,6 @@
+package products
+
+// NullInput is used to indicate to the generic FunctionalTest
+// constructors that the operation being tested does not accept an
+// Input struct
+type NullInput struct{}
```

Comment:
This could probably be moved into `types.go`.

Reply:
I think this is a good idea because it is only needed for testing.

Reply:
Done.